app/models/concerns/orchestration/tftp.rb

```diff
@@ -102,20 +111,14 @@ def delTFTPBootFiles
 
   private
 
-  def template_kind_missing?(kind = host.operatingsystem.template_kind)
-    host.provisioning_template({:kind => kind}).nil?
-  end
-
-  def failure_missing_template
-    failure _("No %{template_kind} templates were found for this host, make sure you define at least one in your %{os} settings") %
-      { :template_kind => host.operatingsystem.template_kind, :os => host.operatingsystem }
-  end
-
   def validate_tftp
     return unless tftp?
     return unless host.operatingsystem
-    return if Rails.env == "test"
-    failure_missing_template if (template_kind_missing? && template_kind_missing?("iPXE"))
+    pxe_kind = host.operatingsystem.pxe_loader_kind(host)
+    if pxe_kind && host.provisioning_template({:kind => pxe_kind}).nil?
+      failure _("No %{template_kind} templates were found for this host, make sure you define at least one in your %{os} settings or change PXE loader") %
```

Comment:
This messsage could also trigger if the host didn't have a `pxe_loader` flag set, right?

 Can you make a different error for that case? I think in that case it's worth pointing out they don't have a `pxe_loader` set for their host.